blog/_posts/2025-08-20-real-time-observability-in-aks-mcp-server.md

```diff
@@ -0,0 +1,195 @@
+---
+title: "Observe Smarter: Leveraging Real-Time insights via the AKS MCP Server"
+description: "Explore how to leverage AKS MCP Server real-time observability to enhance troubleshooting and monitoring of your workloads in AKS clusters."
+date: 2025-08-20
+authors: Qasim Sarfraz
+categories: 
+  - ai
+  - observability
+tags:
+  - ai
+  - mcp
+  - open-source
+  - troubleshooting
+  - inspektor-gadget
+---
+
+## Introduction
+
+The recently released [AKS MCP server](https://blog.aks.azure.com/2025/08/06/aks-mcp-server#option-a-vs-code-extension-recommended) enables AKS customers to automate diagnostics, troubleshooting, and cluster management using natural language.
+One of its key capabilities is real-time observability, which leverages a technology called [eBPF](https://go.microsoft.com/fwlink/?linkid=2259866) to help customers quickly debug applications running in AKS clusters.
+
+> **Note**: If you haven't already seen it, we recently published an [announcement video](https://youtu.be/ZFKgFmkg8jQ?si=eCox_Q8J69MgjqqZ) for the AKS MCP Server
+
+Let’s start with some context. While working in the Linux and Kubernetes observability spaces, we’ve come to better understand challenges customers face, including:
+
+- **Planning troubleshooting**: While there are plenty of tools out there for capturing low-level system and network data, picking the right one and mapping out an effective troubleshooting plan isn’t always easy. It takes some experience and a good understanding of what each tool brings to the table and how to combine them effectively.
+
+- **Capturing real-time data**: Getting real-time data adds another layer of complexity. It often requires specialized tooling and careful coordination so you don’t miss important events as they happen.
+
+- **Data analysis**: While being able to collect all sorts of data is powerful, it comes with the challenge of how to analyze it. Sifting through raw telemetry from multiple sources can quickly become overwhelming, making it hard to extract meaningful insights in a timely way.
+
+Now imagine facing these challenges during a production outage—it just makes everything harder. Even before the recent wave of AI, LLMs, MCPs, etc. we were tackling these observability challenges through an open source project, [Inspektor Gadget](https://go.microsoft.com/fwlink/?linkid=2260072). Now, we can take Inspektor Gadget’s capabilities to the next level through the integration with the AKS MCP server. From here, we’ll walk you through how the `inspektor_gadget_observability` MCP tool, along with other MCP tools in the AKS MCP server, helps reduce these pain points. Let's review this tool's features and then explore some real-world examples.
+
+**Tool**: `inspektor_gadget_observability`
+
+**Available Actions**:
+
+- **deploy**: Deploy Inspektor Gadget to cluster (requires `readwrite/admin` access)
```

Comment:
You can link to https://github.com/Azure/aks-mcp#available-tools for instructions on enabling readwrite/admin access